test/fixtures/use-npm-v10/package.json

```
@@ -0,0 +1,10 @@
+{
+  "name": "use-npm-v10",
+  "version": "1.0.0",
+  "main": "index.js",
+  "license": "MIT",
+  "engines": {
+    "node": "20.x",
+    "npm": "10.x"
```

The issue happens only when `npm` is not specified, and the default npm version is 10.x or greater
```suggestion
    "node": "^20.7"
```

---

@sdaicz it certainly happens when only `node@^20.7` is specified because it defaults to `npm@10.1.0` and the key detail is that the bug is triggered for any npm version greater than 9.  Your suggested setup and mine will both ensure that is the case. I just wanted to be explicit and not rely on the default version that comes with Node to trigger this behaviour during testing.

---

The bug is triggered only if you don't specify `npm`. Because the if condition has `"$version" == ""`. That's what the comment above the if says:
>If the user has not specified a version of npm, but has an npm lockfile
>upgrade them to npm 5.x if a suitable version was not installed with Node

In fact, adding an explicit npm version was our workaround to make the build work with the current version.

---

Apologies @sdaicz, you're right.  Looking back through my changes I started with the unspecified `npm` and then added it to the test fixture later thinking it would be better to make it explicit but that caused the test to report a false positive if the fix was reverted. Using your setup for the test now.